### ISSUE TITLE
Comments/ideas for the post

### ISSUE THREAD

**Issue by maelle** [maelle/niche_packages#5]
* Maintainer of a niche package can stop developing it and if this person is isolated no one will take over.

* How do you reach the 10 potential users of your package i.e. how do you promote it?

* Need to have specific examples, e.g. packages to access a data source nearly no one uses, packages to convert a rare scientific data format, etc. Probably better to use examples from rOpenSci collection (I see this is already noted in the draft).

* Quality linked to having tools making package writing easier (current devtools&co ecosystem), and obviously for in-scope packages, rOpenSci software peer review (that also solves the issue of maintenance if the maintainer quits).

* In ecology a niche doesn't need to be small?

* What is the title of the post. Notes on niche packages? rOpenSci's stance on niche packages?

**Comment by sckott** [comment]
- _... no one will take over_: agreed
- _... how do you promote it_: agree, tough one
- right, an ecological niche doesn't need to be small
- title: not sure yet, i think we should think carefully about whether we want to say anything about ropensci or if we're just speaking in general about R

**Comment by sckott** [comment]
would be good to at least quickly look at the literature to see if this is a topic written about already to see what language is used, any findings related to what we're writing nabout

**Comment by sckott** [comment]
I assume niche does not include packages that are not used widely but ARE general in scope? for example, imagine the package `stringr`, that does something very general, was rarely used, only by a handful of people. would we classify it as niche based on its low usage, or not niche based on its generally wide use case covered?

**Comment by maelle** [comment]
I agreee that niche should relate to the width of possible applications

**Comment by sckott** [comment]
i guess that will be hard to determine in a quantitative way though, yes?  if we had to do manually we'd never get through all the 13K pkgs, though could do manually for ropensci ones

**Comment by maelle** [comment]
Right 🤔

**Comment by maelle** [comment]
I am a bit torn, we could use downloads to illustrate that rOpenSci's collection contains both widely used packages and more confidential ones, and then comment on the idea of niche packages?

Not sure how to make it fit in the post, but it'd be interesting to comment on general on how we evaluate usage of packages (your citation monitoring, the collection of use cases for the newsletters).

**Comment by sckott** [comment]
hmm, i don't think I see how our monitoring of usage fits in, but perhaps I'm not thinking right about it

**Comment by sckott** [comment]
One idea I had:

maybe we can use graphs to get at "niche" quantitatively. that is, use dependency links among packages as a way to determine whether a package is likely to be niche or not. a hypothesis could be:

(NNP: non niche packages; NP: niche packages)

- NNP have many packages depending on on them VS. NP have few packages depending on them
- NNP probably tend to have few dependencies of their own b/c their maintainers want them to be more robust through time VS. NP have many more dependencies as their maintainers aren't aiming for the pkg to be widely depended on by other pkgs

We could calculate graph metrics for these things

Perhaps we could use download data to weight these values perhaps too ?

**Comment by maelle** [comment]
But with this definition you could also get a non-niche but not successful package that'd get classified as NP, e.g. say I write a package for graphics called notggplot2, it's not niche in the sense that I want everyone to use it, but niche in the sense that no one uses it.

That said we can go with your approach and mention limitations at the end :smile_cat:

**Comment by sckott** [comment]
True, good point. we could only include pkgs that have been on CRAN for at least some time period, e.g. 1 year, 2 years, or more. But then you still have the problem of your eg of a non niche pkg being not used, so no dependencies, and then it turns up as niche even though it's not
